### PR TITLE
Use DynamoDB for storage and activate PointInTime backup feature

### PR DESCRIPTION
**Pull request by scisco**
**Summary:** Summary of changes

Addresses [CUMULUS-480](https://bugs.earthdata.nasa.gov/browse/CUMULUS-480)

## Changes

- **CUMULUS-480** Use DynamoDB to store granules, pdrs and execution records
- Activate PointInTime feature on DynamoDB tables
- Increase test coverage on api package
- Add ability to restore metadata records from json files to DynamoDB

## Test Plan
Things that should succeed before merging.

- [x] Unit tests
- [x] Adhoc testing
- [x] Update CHANGELOG

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -11,6 +11,12 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
   - ems.provider property added to `@cumulus/deployment/app/config.yml`. 
     To change the provider name, please add `ems: provider` property to `app/config.yml`.
 
+### Added
+- **CUMULUS-480** Use DynamoDB to store granules, pdrs and execution records
+- Activate PointInTime feature on DynamoDB tables
```
Comment (laurenfrederick) [inline CHANGELOG.md]: These should be indented underneath the first line

File: packages/api/lib/utils.js [hunk 2]
```diff
@@ -19,5 +22,92 @@ function errorify(err) {
...
+
+/**
+ * Returns the collectionId used in elasticsearch
+ * which is a combination of collection name and version
+ *
+ * @param {string} name - collection name
+ * @param {string} version - collection version
+ * @returns {string} collectionId
+ */
+function constructCollectionId(name, version) {
```
[the remark this answers is not in the document]
Comment (scisco) [reply packages/api/lib/utils.js]: I have to remove them from the indexer and use from here.

File: packages/api/endpoints/execution-status.js [hunk 3]
```diff
@@ -10,14 +10,14 @@ const { StepFunction } = require('@cumulus/ingest/aws');
 function get(event, cb) {
   const arn = _get(event.pathParameters, 'arn');
 
-  StepFunction.getExecutionStatus(arn)
+  return StepFunction.getExecutionStatus(arn)
```
Comment (abarciauskas-bgse) [inline packages/api/endpoints/execution-status.js]: Was this causing a bug? (If yes what was the bug?)

File: packages/api/endpoints/execution-status.js [hunk 4]
```diff
@@ -10,14 +10,14 @@ const { StepFunction } = require('@cumulus/ingest/aws');
...
     .then((status) => cb(null, status))
     .catch(cb);
 }
 
 function handler(event, context) {
-  handle(event, context, true, (cb) => {
-    get(event, cb);
+  return handle(event, context, true, (cb) => {
```
Comment (abarciauskas-bgse) [inline packages/api/endpoints/execution-status.js]: Same question here

File: packages/api/endpoints/execution-status.js (hunk 3 above)
Comment (scisco) [reply packages/api/endpoints/execution-status.js]: yeah, if you return a function that has promise inside, you have to make sure that all the other functions with promise also return a promise or the program exists prematurely.

Comment (abarciauskas-bgse) [reply packages/api/endpoints/execution-status.js]: 👍

**Comment by scisco** [comment]
integration tests passed: https://circleci.com/gh/cumulus-nasa/cumulus/2728

**Comment by scisco** [comment]
the `granule-storage` branch is no longer available. Making the PR against master now: https://github.com/cumulus-nasa/cumulus/pull/390

**Pull request closed without merging**